src/main/java/org/truffleruby/core/kernel/KernelNodes.java

```diff
@@ -1610,6 +1612,7 @@ protected boolean doesRespondToSymbol(
                 boolean includeProtectedAndPrivate,
                 @Cached ToJavaStringNode toJavaString) {
```

Comment:
We should really refactor this node to have a single `@Specialization` with a guard of `isRubyString() || isRubySymbol()`, and also call `toJavaString.executeToJavaString(name)` only once.
And simply use `ToSymbolNode` for the `respondToMissing` logic to efficiently convert the name to a Symbol.

If you'd like to do that, it would be very welcome.